pkg/azurefile/azure.go

```diff
@@ -47,6 +48,35 @@ var (
 	storageService = "Microsoft.Storage"
 )
 
+func getRuntimeClassForPod(ctx context.Context, kubeconfig, podName string, podNameSpace string) (string, error) {
+	var (
+		kubeClient *clientset.Clientset
+	)
+	kubeCfg, err := getKubeConfig(kubeconfig, false)
+	if err == nil && kubeCfg != nil {
+		kubeClient, err = clientset.NewForConfig(kubeCfg)
+		if err != nil {
+			klog.Warningf("NewForConfig failed with error: %v", err)
+			return "", err
+		}
+		// Get runtime class for pod
+		if kubeClient != nil {
+			pod, err := kubeClient.CoreV1().Pods(podNameSpace).Get(ctx, podName, metav1.GetOptions{})
+			if err != nil {
+				klog.Warningf("Get pod(%s) failed with error: %v", podName, err)
+				return "", err
+			}
+			if pod.Spec.RuntimeClassName != nil {
+				return *pod.Spec.RuntimeClassName, nil
+			}
+		}
+	} else {
+		klog.Warningf("get kubeconfig(%s) failed with error: %v", kubeconfig, err)
+		return "", err
+	}
+	return "", nil
```

Comment:
Nit: more idiomatic error handling for when we upstream - don't spend time on this if my change doesn't compile. 

```suggestion
	kubeCfg, err := getKubeConfig(kubeconfig, false)
	if err != nil {
		klog.Warningf("getKubeConfig(%s) failed with error: %v", kubeconfig, err)
		return "", err
	}
	var kubeClient *clientset.Clientset
	kubeClient, err = clientset.NewForConfig(kubeCfg)
	if err != nil {
		klog.Warningf("NewForConfig failed with error: %v", err)
		return "", err
	}
	// Get runtime class for pod
	pod, err := kubeClient.CoreV1().Pods(podNameSpace).Get(ctx, podName, metav1.GetOptions{})
	if err != nil {
		klog.Warningf("Get pod(%s) failed with error: %v", podName, err)
		return "", err
	}
	if pod.Spec.RuntimeClassName != nil {
		return *pod.Spec.RuntimeClassName, nil
	}
	return "", nil
```